### DISCUSSION TITLE
How to build project?

### DISCUSSION THREAD

**Discussion by Leo506** [AdamEssenmacher/GoogleApisForiOSComponents#60] [General]
I have followed instructions in `README.md` file:
1. I have installed .net 6 sdk
2. Installed cake and cocoapods
3. Invoked command `dotnet cake --names=Firebase.RemoteConfig --target=externals`

But after all, when I try to call dotnet build on project I get the following error: `error MSB4057: The target "_GetRestoreSettingsPerFramework" does not exist in the project.`

MacOS 15.1.1
XCode 16.1

**Comment by AdamEssenmacher** [comment]
I don't know. I never build projects directly like that--always with `dotnet cake --target=libs`

**Comment by Leo506** [comment]
I have the same error with `dotnet cake --target=libs`:

```
"/Users/user/Projects/GoogleApisForiOSComponents/Xamarin.Google.sln" (Restore target) (1) ->
                   "/Users/user/Projects/GoogleApisForiOSComponents/source/Firebase/Crashlytics/Crashlytics.csproj" (_GenerateRestoreGraphProjectEntry target) (3:9) ->
                   "/Users/user/Projects/GoogleApisForiOSComponents/source/Firebase/Crashlytics/Crashlytics.csproj" (_GetRestoreSettingsPerFramework target) (3:10) ->
                     /Users/user/Projects/GoogleApisForiOSComponents/source/Firebase/Crashlytics/Crashlytics.csproj : error MSB4057: The target "_GetRestoreSettingsPerFramework" does not exist in the project.
```

**Reply by AdamEssenmacher** [reply]
I haven't seen this before. I suspect your machine isn't configured to build the legacy Xamarin target.

Probably not related, but I thought I'd mention too; I also always use the --names=*** parameter to specify which artifacts to build. I haven't brought all of the packages forward from where Microsoft left them and I'm not sure if a 'full' build would work.

**Reply by Leo506** [reply]
Ok, I have fixed this error by manually install Xamarin iOS. But now I have a new error:
`error MSB6006: "bgen" exited with code 1.`

**Reply by AdamEssenmacher** [reply]
What series of commands are you running to get that?

**Reply by Leo506** [reply]
Just `dotnet cake --target=libs --names=Firebase.RemoteConfig `
Here the full log of build:
[buildLog.txt.zip](https://github.com/user-attachments/files/18032694/buildLog.txt.zip)
